### ISSUE TITLE
coastal_irene_atm2roms_intel test is failing under Jenkins workflow

### ISSUE THREAD

**Issue by uturuncoglu** [oceanmodeling/roms#7]
I tried to run `coastal_irene_atm2roms_intel` test under new Jenkins workflow on Hercules and it is failing with following error,

```
20241007 164529.301 ERROR            PET22 ROMS/Master/cmeps_roms.h, ROMS_SetInitializeP2:2138 Object Set or SetDefault method not called  - Passing error in return code
20241007 164529.301 ERROR            PET22 UFS Driver Grid Comp:src/addon/NUOPC/src/NUOPC_Driver.F90:2901 Object Set or SetDefault method not called  - Phase 'IPDv00p2' Initialize for modelComp 2: OCN did not return ESMF_SUCCESS
20241007 164529.301 ERROR            PET22 UFS Driver Grid Comp:src/addon/NUOPC/src/NUOPC_Driver.F90:1980 Object Set or SetDefault method not called  - Passing error in return code
20241007 164529.301 ERROR            PET22 UFS Driver Grid Comp:src/addon/NUOPC/src/NUOPC_Driver.F90:489 Object Set or SetDefault method not called  - Passing error in return code
20241007 164529.301 ERROR            PET22 UFS.F90:394 Object Set or SetDefault method not called  - Aborting UFS
```

The same tests is running fine when it is run manually on same platform. There might be an issue with the ROMS cap or can be related with the environment used by Jenkins.

**Comment by uturuncoglu** [comment]
@hga007 This is part of work related to automated testing environment for UFS Coastal. I know there is not too much information but please let me know if you have any idea. Since it is under Jenkins environment I am not what went wrong but I'll try to debug. @janahaddad We might also need to suggestion from EPIC to debug this case under Jenkins environment or their past experience. Maybe there is hidden bug but not sure.

**Comment by uturuncoglu** [comment]
BTW, I run same test twice to be sure it is not system related and both gave same error.

**Comment by uturuncoglu** [comment]
Tested with GNU and getting same error under GNU compiler

**Comment by uturuncoglu** [comment]
Okay. I track this issue to initial() call at https://github.com/myroms/roms/blob/5a15936cc4a307e3100c52ee209a593b463856e5/ROMS/Drivers/nl_roms.h#L209

**Comment by hga007** [comment]
@uturuncoglu: Okay, so something in the **initial** routine is making it fail.  However, we need more information to target the issue with that particular system.  It is a very short routine.  I suggest putting PRINT statements to see what call inside **initial** failed.  Then, we could figure out what is going on.

**Comment by uturuncoglu** [comment]
@hga007 yes. I am tracking more deep at this point. I'll update you once I found the exact place.

**Comment by uturuncoglu** [comment]
@hga007 It seems that model is crashing in the following line under Jenkins workflow,

https://github.com/oceanmodeling/roms/blob/15ae96312ff76bc5f58b91f637310fdf9adfb07b/ROMS/Nonlinear/initial.F#L502

I am not sure why but I could add more debug prints inside of `obs_initial`. Ar you seeing any obvious error around here,

https://github.com/oceanmodeling/roms/blob/15ae96312ff76bc5f58b91f637310fdf9adfb07b/ROMS/Nonlinear/initial.F#L490

In this case, the last print that see is with 16 and could not see 17. I am also seeing following in the error log. I have also following error in the log,

/role-nosofs/FV3_RT/rt_3453913/coastal_irene_atm2roms_intel/log.roms 

 OBS_INITIAL_NF90 - unable to open input NetCDF file: /work2/noaa/nems/tufuk/RT/NEMSfv3gfs/input-data-20240501/ROMS/IRENE/Data/OBS/irene_obs_20110827.nc

but the file is in there and I could see its content with ncdump. Maybe it is just a permission issue. BTW, the model copies those files to run directory but still points to `OBSname == /work2/noaa/nems/tufuk/RT/NEMSfv3gfs/input-data-20240501/ROMS/IRENE/Data/OBS/irene_obs_20110827.nc` in `rbl4dvar.in`. Maybe this scripts needs to be adjusted.

https://github.com/oceanmodeling/ufs-weather-model/blob/711f55369d880c32548465a2443b4f31bc468df8/tests/fv3_conf/coastal_roms_run.IN#L11

It is using https://github.com/myroms/roms_test/blob/main/IRENE/Coupling/roms_data_cmeps/job_setup.sh

**Comment by hga007** [comment]
@uturuncoglu: The error tells us we cannot open the **`irene_obs_20110827.nc`** NetCDF. I was able to look at that file. Can you do a **ncdump**? Maybe that file is corrupt. Did you set the LFS in Git Hub configuration when downloading that file?

I tried **`ncks --chk_nan irene_obs_20110827.nc`**, and I didn't find NaN in the file.

**Comment by uturuncoglu** [comment]
@hga007 The data seems fine. I check it without command. I think 

https://github.com/myroms/roms_test/blob/a66d2efd66092f349ea0f00624c1e7a7d2bf683c/IRENE/Coupling/roms_data_cmeps/rbl4dvar.tmpl#L419

paths needs to point the run directory but at this point they are pointing input directory. I'll try to play around to see if I could run the case.

**Comment by uturuncoglu** [comment]
@hga007 Okay. If I do following change in the template file, the case is running under Jenkins fine.

        !OBSname == MyIRENEdir/Data/OBS/irene_obs_20110827.nc
        OBSname == irene_obs_20110827.nc

So, I think we need to revisit template files and also `job_setup.sh` scripts to look the files in the run directory rather than using common input directory. It seems that Jenkins role account does not like it. Maybe we could introduce another option to `job_setup.sh` to force to use run directory and the could use that option only in UFS Coastal RT script and this will not change the default behavior for running roms cases outside of the UFS Coastal. Do you think that it is easy to do that? Let me know if you have any other suggestion.

**Comment by uturuncoglu** [comment]
@hga007 I think that `IreneDir` in the script needs to provide entire folder rather then pointing top level folder. BTW, if we set `IreneDir` to current directory `.` then it would create templates based on it. Maybe we need to have two directory. One is source (will point actual data path) and another one is destination (it is run directly and also used to create templates). So, with this approach you could set both same outside of the UFS Coastal and it works as it is but under UFS Coastal RT we could set it to run directory.

**Comment by uturuncoglu** [comment]
@hga007 I could also go and modify the files manually used by UFS Coastal but then once roms_test repository updates them. We have to do it again. I am not sure what is the best approach in here.

**Comment by hga007** [comment]
@uturuncoglu: Perhaps, you could do an **`ln -s`** to that file in the running directory. You can also turn off the **`VERIFICATION`** CPP option; you will not need that file. It creates a station file at the observation location, which is unnecessary for the UFS workflow.

**Comment by uturuncoglu** [comment]
@hga007 I am already copying obs file to run directory but script generated `rbl4dvar.in` file still looks main input directory not run directory. Okay for short term solution I'll disable `VERIFICATION` flag and test it but in longer term this could create potential issue with other cases.

**Comment by uturuncoglu** [comment]
@hga007 Okay. I disabled verification flag and it is working fine under Jenkins. I am plaining to close this ticket but it would be nice to have flexibility to generate those namelist files to look at run directory rather than looking to main input directory. Let me know what you think.

**Comment by uturuncoglu** [comment]
@janahaddad JFYI, here is the screen shout of successful run. 

![Image](https://github.com/user-attachments/assets/cecd2879-9111-44f1-9076-0636f3f5d638)

I am not sure why role account could not read data in `/work2/noaa/nems/tufuk/RT/NEMSfv3gfs/input-data-20240501/ROMS/IRENE/Data/OBS/irene_obs_20110827.nc`. You might want to check it with someone that created the role account. At this point, we have a workaround (by deactivating verification with this commit https://github.com/oceanmodeling/ufs-weather-model/commit/6b1a4690ebf44d5064ea2505c2a63bc2f6369d66) but this is not a long terms solution and it would be nice to check the role account. Anyway, maybe it would be nice to see this ticket open for follow up the issue.

**Comment by uturuncoglu** [comment]
@janahaddad JFYI, GNU is also fine with this change.

**Comment by uturuncoglu** [comment]
@janahaddad This is already solved. I am closing this ticket.
